torch/distributed/_sharded_tensor/api.py

```diff
@@ -545,11 +559,13 @@ def sharding_spec(self) -> ShardingSpec:
         return self._sharding_spec
 
     def __torch_function__(self, func, types, args=(), kwargs=None):
-        if func == torch.nn.functional.linear:
+        if func in _CUSTOM_SHARDED_OPS:
+            return _CUSTOM_SHARDED_OPS[func](types, args, kwargs, self._process_group)
+        elif func == torch.nn.functional.linear:
```

Comment:
Do you feel we should always use this decorator for all new ops and migrate the existing ones using this decorator, too?

Reply:
Yes, this indeed might be much more cleaner. Let me update the PR.